src/compliment/utils.clj

```diff
@@ -217,3 +217,39 @@
                            (.endsWith file ".jar") (.endsWith file ".class")))]
         (if (.startsWith file File/separator)
           (.substring file 1) file)))))
+
+
+(defn as-sym [x]
+  (if x (symbol x)))
+
+(defn namespace-sym
+  "Return the namespace of a fully qualified symbol if possible.
+
+  It leaves the symbol untouched if not."
+  [sym]
+  (if-let [ns (and sym (namespace sym))]
+    (as-sym ns)
+    sym))
+
+(defn name-sym
+  "Return the name of a fully qualified symbol if possible.
+
+  It leaves the symbol untouched if not."
+  [sym]
+  (if-let [n (and sym (name sym))]
+    (as-sym n)
+    sym))
+
+(defn unquote-first
+  "Handles some weird double-quoting in the analyzer"
+  [form]
+  (if (= (first form) 'quote)
+    (first (rest form))
+    form))
+
+(defn normalize-arglists
+  [arglists]
```

Comment:
Could be useful to have a docstring for this function, otherwise it's hard to understand why exactly it is needed.